### PR TITLE
Add compact JSDoc Rule

### PR DESCRIPTION
**Pull request by ianwremmel**
This a (hopefully) temporary measure until the [PR against tslint](https://github.com/palantir/tslint/pull/3925) is accepted.

Note: that this treats improperly formatted JSDocs as errors; if we cannot autofix those errors, we should override on a per-repo basis to reduce severity to "warning".

### CODE REVIEW

File: src/ud/jsdocFormatRule.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    // intentionally empty
+  }
+
+  const lines = fullText.slice(startPos, end).split('\n');
+
+  if (lines.length !== 1) {
+    return false;
+  }
+
+  if (lines[0].length <= 80) {
```
Comment (jbraithwaite) [inline src/ud/jsdocFormatRule.ts]: Should this be configurable?

Comment (kourge) [reply src/ud/jsdocFormatRule.ts]: As I understand it, this PR is a temporarily workaround, for which the more general solution is https://github.com/palantir/tslint/pull/3925. The better solution doesn't seem to have this hardcoded value.

Comment (ianwremmel) [reply src/ud/jsdocFormatRule.ts]: yea, i figured it could be made configurable at some point, but since we use 80 anyway, i wasn't that worried about it.

**Pull request merged**